pkg/cache/cache.go

```diff
@@ -562,6 +651,7 @@ func (c *Cache) UpdateWorkload(oldWl, newWl *kueue.Workload) error {
 	if !ok {
 		return fmt.Errorf("new ClusterQueue doesn't exist")
 	}
+	c.broadcastToRecheckPodsReadyForUpdatedWorkload(oldWl, newWl)
```

Comment:
I want to broadcast only if it makes sense to recheck the condition - meaning that there was an admitted workload with PodsReady=False and now the workload has PodsReady=True or is gone. Rechecking the condition on adding a new workload does not seem necessary.

Reply:
+1 to inlining here

Reply:
Back to this:

> Also you probably need to check when the Workload is first added

when we admit a workload, it could immediately start running and become ready before kueue gets the event that runs `AddOrUpdateWorkload`. So you need to check if the pods are ready and broadcast in that function too.

Reply:
Done